PluginGenerator/JarManifestBuilder.cs

```diff
@@ -0,0 +1,162 @@
+            }
+
+            sb.AppendLine();
+
+            return sb.ToString();
+        }
+
+        /// <summary>
+        /// Writes the manifest file to the specified directory
+        /// </summary>
+        public string WriteManifest(string directory)
+        {
+            if (string.IsNullOrWhiteSpace(directory))
+            {
+                throw new ArgumentNullException(nameof(directory));
+            }
+
+            // Ensure the directory exists
+            Directory.CreateDirectory(directory);
+            string filePath = Path.Combine(directory, ManifestFileName);
+
+            File.WriteAllText(filePath, this.GetManifest());
+
+            return filePath;
+        }
+
+        #endregion
+
+        #region Private methods
+
+        private static void CheckAttributeNameIsValid(string name)
+        {
+            // [0-9a-zA-Z_-]
+            // 70 chars max
+            if (string.IsNullOrWhiteSpace(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (name.Length > MaxNameLength)
+            {
+                string message = string.Format(System.Globalization.CultureInfo.CurrentCulture, UIResources.JMan_Error_NameTooLong, name);
+                throw new ArgumentException(message, nameof(name));
+            }
+
+            if (!System.Text.RegularExpressions.Regex.IsMatch(name, ValidNamePattern))
+            {
+                string message = string.Format(System.Globalization.CultureInfo.CurrentCulture, UIResources.JMan_Error_InvalidCharsInName, name);
+                throw new ArgumentException(message, nameof(name));
+            }
+        }
+
+        /// <summary>
+        /// Writes an entry, splitting it across multiple lines if required
+        /// </summary>
+        /// <remarks>NOTE: we assuming there are no multi-byte characters in the value. If there are,
+        /// the string won't be split correctly</remarks>
+        private static void WriteEntry(string key, string value, StringBuilder sb)
+        {
+            string text = string.Format(System.Globalization.CultureInfo.InvariantCulture, "{0}: {1}", key, value);
+
+            Debug.Assert(!value.Any(c => char.IsSurrogate(c)), "Not expecting multi-byte characters");
+
+            if (text.Length <= MaxLineLength)
+            {
+                sb.AppendLine(text);
+                return;
+            }
+
+            // Data is too long a single line
+            // Write the first line without a continuation character
+            sb.AppendLine(text.Substring(0, MaxLineLength));
+            text = text.Substring(MaxLineLength);
+
+            // For every remaining line, we need to reserve characters
+            // for the continuation prefix
+            const string continuationPrefix = " ";
+            int maxActualCharsPerLine = MaxLineLength - continuationPrefix.Length;
+            
+            while (text.Length > maxActualCharsPerLine)
```

Comment:
it should be possible to change this condition into `while (!text.IsEmpty()) { .. }` and get rid of the `Write out whatever is left` below